File: src/main/java/ru/anseranser/peshki/engine/GameConfig.java
package ru.anseranser.peshki.engine;

public record GameConfig(
        int sideLength,
        int numberOfPlayers,
        int numberOfPawns,
        int numberOfDice,
        int numberOfSidesOnDice,
        int maxTurns
) {
    public static final GameConfig DEFAULT = new GameConfig(8, 4, 4, 2, 6, 5000);

    public int fieldLength() {
        return numberOfPlayers * (sideLength - 2);
    }

    public int homeLength() {
        return numberOfPawns - 1;
    }
}

File: web/src/engine/GameConfig.ts
export interface GameConfig {
  sideLength: number;
  numberOfPlayers: number;
  numberOfPawns: number;
  numberOfDice: number;
  numberOfSidesOnDice: number;
  maxTurns: number;
}

export const DEFAULT_CONFIG: GameConfig = {
  sideLength: 8,
  numberOfPlayers: 4,
  numberOfPawns: 4,
  numberOfDice: 2,
  numberOfSidesOnDice: 6,
  maxTurns: 5000,
};

export function fieldLength(config: GameConfig): number {
  return config.numberOfPlayers * (config.sideLength - 2);
}

export function homeLength(config: GameConfig): number {
  return config.numberOfPawns - 1;
}
